technical-reference/reference-technical-articles/technical-doc-token-converter.md

```diff
@@ -0,0 +1,226 @@
+    - In this case, there isn’t enough liquidity to use 100% of the input amount (25 XVS would be 126,25 USDC), so 100% of the liquidity will be the `amountOutMantissa`, and this function (`getAmountOut`) calculates the input amount needed to “cover” that amount (taking into account the discount)
+    - Response:
+        - `amountConvertedMantissa`: ((100 / 1,01) / 5) * 10^18 = **19,80 XVS**, where:
+            - 100 is the liquidity of USDC in the contract
+            - 1,01 is the application of the discount
+            - 5 is the conversion rate given by the oracles
+        - `amountOutMantissa`: 100 * 10^18. That is **100 USDC**
+
+### Get amount in
+
+View function `getAmountIn(uint256 amountOutMantissa, address tokenAddressIn, address tokenAddressOut) returns (uint256 amountInMantissa, uint256 amountConvertedMantissa)`, to request the amount of `tokenAddressIn` tokens that a sender should send to the contract, to receive `amountOutMantissa` tokens of `tokenAddressOut`, where:
+
+- Parameters:
+    - `amountOutMantissa`: the amount of `tokenAddressOut` tokens the sender would like to receive in the conversion. It is defined in terms of the mantissa of `tokenAddressOut`
+    - `tokenAddressIn`: the address of the token provided by the sender to get tokens of `tokenAddressOut`
+    - `tokenAddressOut`: the address of the token to get after the conversion
+- Returned values:
+    - `amountInMantissa`: the amount of `tokenAddressIn` tokens that the sender has to send to the contract to receive `amountConvertedMantissa` tokens of `tokenAddressOut`
+    - `amountConvertedMantissa`: the amount of `tokenAddressOut` tokens that would be really transferred from the contract to the sender if the user sends `amountInMantissa` tokens of `tokenAddressIn` to the contract. If there is enough liquidity of `tokenAddressOut` in the contract, this amount is equal to `amountOutMantissa`. Otherwise, it will be less than the amount
+
+Examples:
+
+- Invocation 1: `getAmountIn(40 * 10^18, address(XVS), address(USDC))`
+    - The configuration enables for `tokenAddressIn / tokenAddressOut`: XVS/USDC
+    - Current liquidity: 100 USDC
+    - Current conversion rate, given by the oracle: 1 XVS = 5 USDC
+    - Configured discount: 0%
+    - Response:
+        - `amountInMantissa`: 8 * 10^18 (XVS)
+        - `amountConvertedMantissa`: 40 * 10^18 (USDC), equal to the `amountOutMantissa` param
+- Invocation 2: `getAmountIn(120 * 10^18, address(XVS), address(USDC))`
+    - The configuration enables for `tokenAddressIn / tokenAddressOut`: XVS/USDC
+    - Current liquidity: 100 USDC
+    - Current conversion rate, given by the oracle: 1 XVS = 5 USDC
+    - Configured discount: 0%
+    - Response:
+        - `amountInMantissa`: 20 * 10^18 (XVS)
+        - `amountConvertedMantissa`: 100 * 10^18 (USDC), different from the `amountOutMantissa` param because there is not enough liquidity
```

Comment:
```suggestion
        - `amountConvertedMantissa`: $100 * 10^{18}$ (USDC), different from the `amountOutMantissa` param because there is not enough liquidity
```